### PR TITLE
move steps around to create largest overlapping sections

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: .github/workflows/test-regression.yaml [hunk 1]
```diff
@@ -169,6 +174,45 @@ jobs:
           make -j $(nproc) ${{ inputs.mode }} 2>&1 | tee ${{ steps.build-log.outputs.build_log_name }}.txt
           set +o pipefail
 
+
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:177]: Extraneous space

File: .github/workflows/test-regression.yaml [hunk 2]
```diff
@@ -169,6 +174,45 @@ jobs:
...
+          path: |
+            riscv-gnu-toolchain/stage2.zip
+          retention-days: 5
+
+      - name: Output success
+        id: build-success
+        run: |
+          export BUILD_SUCCESS="success"
+          echo $BUILD_SUCCESS
+          echo "build_success=$BUILD_SUCCESS" >> "$GITHUB_OUTPUT"
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:212]: If I'm gonna start uploading the build logs in my repo too, it probably makes sense to move:
Zip build log
Upload build log
into common as well. Or at least move those above Build success report so all the comment-related tasks are grouped together.

File: .github/workflows/test-regression.yaml [hunk 3]
```diff
@@ -925,6 +948,9 @@ jobs:
           python scripts/post_check_to_patchworks.py -event ${{ github.event_name }} -repo ewlu/gcc-precommit-ci -pid $(tail -n 1 patches/${{ inputs.patch_name }})  -desc 'Testing started' -iid '${{ inputs.issue_num }}#issuecomment-${{ inputs.test_comment_id }}' -state 'pending' -context 'test' -token 'PLACEHOLDER'
         continue-on-error: true
 
+
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:951]: Stray space

File: .github/workflows/test-regression.yaml [hunk 4]
```diff
@@ -803,6 +824,20 @@ jobs:
...
+      - name: Upload binary artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: gcc-${{ inputs.mode }}-${{ inputs.target }}-${{ inputs.patch_applied_gcchash }}-${{ inputs.multilib }}
+          path: |
+            riscv-gnu-toolchain/gcc-${{ inputs.mode }}-${{ inputs.target }}-${{ inputs.patch_applied_gcchash }}-${{ inputs.multilib }}.zip
+          retention-days: 5
+
+      ### end common
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:839]: Same thing here about uploading the build log immediately so the comment related stuff is all grouped together

File: .github/workflows/test-regression.yaml [hunk 5]
```diff
@@ -925,6 +948,9 @@ jobs:
           python scripts/post_check_to_patchworks.py -event ${{ github.event_name }} -repo ewlu/gcc-precommit-ci -pid $(tail -n 1 patches/${{ inputs.patch_name }})  -desc 'Testing started' -iid '${{ inputs.issue_num }}#issuecomment-${{ inputs.test_comment_id }}' -state 'pending' -context 'test' -token 'PLACEHOLDER'
         continue-on-error: true
 
+
+      ### begin common
```
Comment (patrick-rivos) [inline .github/workflows/test-regression.yaml:952]: This is the same as a section in rerun-timeouts. Can we mark that section as well so we share between these?

**Review by patrick-rivos — CHANGES_REQUESTED** [review]

File: .github/workflows/test-regression.yaml (hunk 5 above)
Comment (ewlu) [reply .github/workflows/test-regression.yaml:952]: https://github.com/ewlu/gcc-precommit-ci/pull/480/files/169250d5d8677a7765954d14004ab37119ead30d#diff-65513f4ceb3e748a59193e4a9502e06bc5ed1cb466247169f89c63460db7cfc2R588-R589

File: .github/workflows/test-regression.yaml (hunk 2 above)
Comment (ewlu) [reply .github/workflows/test-regression.yaml:212]: moved above build success report

File: .github/workflows/test-regression.yaml (hunk 4 above)
Comment (ewlu) [reply .github/workflows/test-regression.yaml:839]: moved above build success report

**Review by patrick-rivos — APPROVED** [review]
LGTM

**Pull request merged**